app/models/enhancements/artefact.rb

```diff
@@ -8,6 +8,11 @@ class Artefact
   # without persisting it
   attr_accessor :indexable_content
 
+  # Add a non-field attribute so we can pass the redirect_url over to the Router
+  # without persisting it
+  attr_accessor :redirect_url
```

Comment:
I think we should be storing the redirect URL.  Otherwise we have no way to re-register these artefacts, and run the risk of the redirect being replaced with a gone route.